### ISSUE TITLE
Can the token be parsed after expiration?

### ISSUE THREAD

**Issue by ghost** [jwtk/jjwt#183]
Hi
I have a custom claims object built into my token and I was wondering if there is a way to Parse the token and skip the time expiration check? This will help me troubleshoot any production issues long after the token has expired.

**Comment by Hazem-Ben-Khalfallah** [comment]
I'm interested to get this question's response. I need to retrieve claims from an expired token and I don't see how to do it. Can this be done?

**Comment by nvasilescu** [comment]
I face the same problem. I'm using it fora  backend for a mobile app. If the token has expired, I need to generate a new one belonging to the same user. The claims from the expired token can't be read giving expired token exception. I have now 2 options theoretically, practically just one. To make the token expire in a very long period. You can't force a user to login on mobile because it's not the right way to do it.

**Comment by Hazem-Ben-Khalfallah** [comment]
I have found a way to retrieve claims but without using jjwt code. Here is what I used

`public String parse(String token) { 
       final String encodedPayload = token.split("\\.")[1]
                .replace('-', '+').replace('_', '/'); 
       final String decoded = new String(Base64.getDecoder().decode(encodedPayload));
  }`

This method will return a string containing a serialized Json with Jwt token claims. Then It can be parsed using any Json parser like Jackson and retrieve all the values that you need.

**Comment by kamilwlf** [comment]
Can this be done out of the box?

**Comment by psevestre** [comment]
If you don´t have a "not-before" claim, then you can set the Clock instance used by the DefaultJwtParser. In your case, the Clock implementation would always return a date far,far in the future so no reasonable token would be expired.

**Comment by kamilwlf** [comment]
I added, it works:

```
 .setClock(new Clock() {
                        @Override
                        public Date now() {
                            return new Date(System.currentTimeMillis()/2);
                        }
```

Did you write about it?

**Comment by mikegleasonjr** [comment]
@Hazem-Ben-Khalfallah You are not verifying the token signature, so you can accept easily forged tokens.

@psevestre, @kamilwlf  This really is a hack.

I will have to move away from this library since I too need to verify a token signature independently of its expiration date. 

I need to refresh the access token using a refresh token, but I want to make sure someone had a valid access token beforehand (expired or not).

**Comment by IldarFattakhov** [comment]
I think this is ок:
`              try 
	        {
	            Jws<Claims> jwsR = Jwts.parser().setSigningKey( key ).parseClaimsJws( jws );
	            System.out.println( "value = " + jwsR.getBody().get( "value" ) );
	        }    
	        catch ( ExpiredJwtException ex ) {
	            System.out.println( "value = " + ex.getClaims().get( "value" ) );
	        }`

**Comment by lhazlewood** [comment]
This will be resolved via #148.  Closing as a duplicate of that issue.
